### PR TITLE
[4.14] Add deprecation warnings on {Int32,Int64,Nativeint}.format

### PR DESCRIPTION
**Pull request by nojb**
These (unsound) functions have been removed from 5.0 in #10867. In this PR (based on 4.14) we add the necessary deprecation warnings for 4.14.

cc @Octachron

### CODE REVIEW

**Comment by xavierleroy** [comment]
> These (unsound) functions have been removed from 5.0 in #10867

No, they have not (not in #10867 and not in another PR as far as I can see) but they should be real soon now.

File: testsuite/tests/basic/boxedints.ml
Comment (xavierleroy) [inline testsuite/tests/basic/boxedints.ml:123]: It would be nice to keep these tests (they exercise conversions from hex strings to integers, not just from integers to hex strings) but make them use Printf.sprintf instead.

Comment (nojb) [reply testsuite/tests/basic/boxedints.ml:123]: Good idea. I pushed a commit moving these tests to `lib-printf/tprintf.ml`.

**Comment by nojb** [comment]
> > These (unsound) functions have been removed from 5.0 in #10867
> 
> No, they have not (not in #10867 and not in another PR as far as I can see) but they should be real soon now.

Yes, they were (it was an oversight on my part, so they are not listed in the `Changes` entry):

https://github.com/ocaml/ocaml/pull/10867/files#diff-d648208b6b2b71f3080ef737cc7055a98e774f788314b685640c7e612a66d5b3

Once this PR is merged, I will make a PR against 5.0 fixing the `Changes` entry and also updating the tests for `Printf.sprintf` as was done here.

File: testsuite/tests/lib-printf/tprintf.ml
Comment (Octachron) [inline testsuite/tests/lib-printf/tprintf.ml:616]: The tests for `Nativeint` seem to be missing. However, they could be added later: I am quite keen to have a version of 4.14 with all new deprecation warnings.

Comment (nojb) [reply testsuite/tests/lib-printf/tprintf.ml:616]: Yes, there are no tests for `Nativeint`, as they are harder to write (the tests depend on whether you are on 32- or 64-bits). I wasn't planning to add them...

https://github.com/ocaml/ocaml/blob/3aaa9a7cf52b6980a850bfafde724157a783b98d/testsuite/tests/lib-printf/tprintf.ml#L515-L516

All that is missing to merge this PR is an official approval :)

**Review by Octachron — APPROVED** [review]
I am approving those changes for 4.14: those functions are unsafe and has been removed from the user-visible documentation for nearly 20 years.

It is time to remove them in OCaml 5. Consequently, if we want that any programs that compile without deprecation warnings in 4.14 transfers to OCaml 5 without issues, we need to deprecate them in 4.14

**Comment by nojb** [comment]
Thanks! Merged.

**Pull request merged**
